Release/17.0.0

gzunigal opened the pull request:
Implements ETL for gathering information of medical centers

* Accepts a patient information json through an API
* Queue the request for processing
* Transform the request's data to a standardized schema
* Persists the information sending the data to Unicorns' API

gzunigal committed:
Merge pull request #3 from EtherCorp/feature/project-setup

Setup project

gzunigal committed:
Adds postgresql support and configuration

gzunigal committed:
Removes Active Records

This commit removes AR because the DB will be accessed from the unicorn main project.
Obs: MongoDB driver needed

gzunigal committed:
README.md updated

lmigryk committed:
Merge pull request #5 from EtherCorp/feature/project-setup

Base project structure and configuration

gzunigal committed:
gemlock

gzunigal committed:
Created structure of API

- Versioning support
- Includes an example endpont

gzunigal committed:
Implemented a simple activity logger

- Logger configuration in lib/actyvity_logger.rb
- USE: ActivityLogger.log({log_type: 'Test', message: 'Test Message'})

sacranister committed:
Merge pull request #7 from EtherCorp/feature/API

Approved

gzunigal committed:
Merge pull request #6 from EtherCorp/feature/logger

Feature/logger

israel-martinez committed:
implement json queue ENDPOINT with sidekiq (redis implementation)and save the data in mongodb with mongoid (ruby gem)

lmigryk committed:
added faker folder

kevin-alvarez committed:
gem http added, first attempt request

kevin-alvarez committed:
GET POST DELETE and patient Request

sacranister committed:
Request implementation

sacranister committed:
JSON received by faker-data stored in MongoDB

sacranister committed:
POST format fixed

sacranister committed:
fields added to JSON

JavieraSaezLeon committed:
se estandarizan dos tipos de json, pero aun no se genera json final.... se crea main para pruebas

gzunigal committed:
Merge pull request #8 from EtherCorp/feature/faker-data

Feature/faker data

JavieraSaezLeon committed:
return of object

israel-martinez committed:
concept test workers sidekiq (Polanco, Martínez)

JavieraSaezLeon committed:
add two new values in object returned

Commit:
update proyect with hf

sacranister committed:
fields added to JSON fixed

sacranister committed:
Merge branch 'develop' into feature/API-communication
need of faker-app in this branch

sacranister committed:
token field added in faker

kevin-alvarez committed:
requests for all entities

kevin-alvarez committed:
Request module and JSON parse for POST

kevin-alvarez committed:
Change params to recieve JSON object

gzunigal committed:
Merge branch 'develop' into feature/Drivers

sacranister committed:
some Rubocop fixs

gzunigal committed:
Base structure of Driver module

gzunigal committed:
Faker App refactored

israel-martinez committed:
Merge pull request #12 from EtherCorp/feature/requests

Feature/requests merge to Develop

gzunigal committed:
Add Readme

israel-martinez committed:
save current changes, workers test consumer enpoints

israel-martinez committed:
Merge branch 'develop' into feature/queue

Conflicts:
	centers-api/Gemfile

gzunigal committed:
Basic driver implementation
- Driver::MedicalCenter1 parse a MedicalCenterFaker's patient request
- Includes a README

gzunigal committed:
Minor changes

- Tagged old driver as deprecated and saved for reference
- Parsed rut as String

gzunigal committed:
Update include

sacranister committed:
new Faker

gzunigal committed:
Fixed bad URL

gzunigal committed:
Fix in rut parsing

gzunigal committed:
Fixed uri concatenation

sacranister committed:
Merge pull request #13 from EtherCorp/feature/faker-refactoring

Faker refactoring

sacranister committed:
change endpoint / to /patients

israel-martinez committed:
Add base to custom_modules.rb for global access, testing Request

sacranister committed:
change name of collection in MongoDB

sacranister committed:
fix FAKER problem

israel-martinez committed:
chargue all request from lib path with custom_modules.rb

israel-martinez committed:
sandbox branch, create workers and send request to u_project

Commit:
Define class DispatcherWorker that send request to unicorn with sidekiq

israel-martinez committed:
Merge pull request #10 from EtherCorp/feature/Drivers

Feature/drivers

israel-martinez committed:
Merge remote-tracking branch 'origin/feature/workers' into feature/workers

Conflicts:
	centers-api/config/initializers/custom_modules.rb

israel-martinez committed:
update custom_modules.rb

gzunigal committed:
Merge pull request #14 from EtherCorp/feature/API-communication

Feature/api communication

israel-martinez committed:
Merge branch 'develop' into feature/workers

israel-martinez committed:
Integrate flow: FAKER-> API(PERSISTENCE)-> WORKERS: (DRIVER)

israel-martinez committed:
Integrate flow: FAKER-> API(PERSISTENCE)-> WORKERS: (DRIVER)

israel-martinez committed:
Merge remote-tracking branch 'origin/feature/workers' into feature/workers

Conflicts:
	centers-api/app/api/v1/base.rb
	centers-api/app/workers/dispatcher_worker.rb
	centers-api/config/routes.rb

israel-martinez committed:
Minor changues

israel-martinez committed:
fix

gzunigal committed:
Implemented data persistence through Unicorn API

JavieraSaezLeon committed:
Merge pull request #15 from EtherCorp/feature/workers

Use DispatcherWorker

JavieraSaezLeon committed:
README

define how to set up and run

JavieraSaezLeon committed:
Update README.md

JavieraSaezLeon committed:
Update README.md

gzunigal committed:
Merge pull request #16 from EtherCorp/JavieraSaezLeon-patch-1

README

gzunigal committed:
Merge pull request #17 from EtherCorp/develop

Develop

Commit:
gemfile versions fixed

(cherry picked from commit 6bb5205)

dochoaj commented on centers-api/Gemfile:
```diff
@@ -39,3 +37,22 @@ end
+
+
+# Gemfile
+gem 'grape'
```
Fijar las versiones de las gemas

dochoaj commented on centers-api/app/api/v1/base.rb:
```diff
@@ -0,0 +1,34 @@
+      }
+      conn.save_patient(patient)
+      request_type = 'patients'
+      puts '************************'
```
Hay maneras de crear logs sin usar puts.

dochoaj commented on centers-api/app/controllers/jsons_controller.rb:
```diff
@@ -0,0 +1,40 @@
+class JsonsController < ApplicationController
```
No usemos controllers, usemos Grape para manejar parámetros JSON

dochoaj commented on centers-api/app/models/json.rb:
```diff
@@ -0,0 +1,8 @@
+# For json data from medical_center
+class Json
```
Le cambiaría el nombre a la clase.

dochoaj commented on centers-api/app/requests/request/consult.rb:
```diff
@@ -0,0 +1,20 @@
+    @entity = "consults"
+
+    def self.GET_movement(id)
+      dir = @@url + @entity + "/" + id.to_s + "/" + "movements"
```
Cuando los strings no necesitan interpolación, se usa comilla simple.

dochoaj commented on centers-api/app/requests/request/speciality.rb:
```diff
@@ -0,0 +1,7 @@
+module Request
+  class Speciality < Request::Base
+    @entity = "specialities"
+
```
Muchos espacios innecesarios

dochoaj commented on centers-api/app/workers/dispatcher_worker.rb:
```diff
@@ -0,0 +1,23 @@
+
+    puts 'end transform data to Unicorn with'
+    Request::Patient.POST(normalized_json)
+    puts 'TODO: Is saved data?'
```
Puts ToDO?

dochoaj commented on centers-api/bin/setup:
```diff
@@ -19,14 +19,6 @@ chdir APP_ROOT do
   system('bundle check') || system!('bundle install')
 
 
-  # puts "\n== Copying sample files =="
```
Por que es esto???

dochoaj commented on centers-api/bin/setup:
```diff
@@ -19,14 +19,6 @@ chdir APP_ROOT do
 
 
-  # puts "\n== Copying sample files =="
-  # unless File.exist?('config/database.yml')
```
Y esto?

dochoaj commented on centers-api/config/mongoid.yml:
```diff
@@ -0,0 +1,157 @@
+development:
```
Esto debería ir gitignoreado, tal cual les comenté en el otro repo.

dochoaj reviewed (changes requested):
Les dejé comentarios!

gzunigal committed:
README updated and minor fixes

- README now guide step by step the set up and run process
- API uses custom logger
- Requests module uses string interpolation
- Worker logs has been cleaned

gzunigal committed:
Merge remote-tracking branch 'origin/release/17.0.0' into release/17.0.0

gzunigal commented:
@dochoaj Se arregló todo lo que comentó
